# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1260426506 18000
#      Thu Dec 10 01:28:26 2009 -0500
# Node ID 33537a4d223ff5d0e00fcecdd7ca3afa97f1009e
# Parent  de6b6173c99033e55eb756ac203e4b273f9c6408
mkoctfile.cc.in: accept long options with single -

diff --git a/ChangeLog b/ChangeLog
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,8 +1,12 @@
+2009-12-10  John W. Eaton  <jwe@octave.org>
+
+	* mkoctfile.cc.in: Accept long options with single -.
+
 2009-12-10  John W. Eaton  <jwe@octave.org>
 
 	* mkoctfile.in: Accept long options with single -.
 
 2009-12-09  John W. Eaton  <jwe@octave.org>
 
 	* HACKING: Clarify gnulib checkout and use.
 
diff --git a/mkoctfile.cc.in b/mkoctfile.cc.in
--- a/mkoctfile.cc.in
+++ b/mkoctfile.cc.in
@@ -408,17 +408,19 @@ main (int argc, char **argv)
   bool compile = true;
 
   if (argc == 1)
     {
       cout << usage_msg << endl;
       return 1;
     }
 
-  if (argc == 2 && (!strcmp (argv[1], "-v") || !strcmp (argv[1], "--version")))
+  if (argc == 2 && (!strcmp (argv[1], "-v")
+                    || !strcmp (argv[1], "-version")
+                    || !strcmp (argv[1], "--version")))
     {
       cout << version_msg << endl;
       return 0;
     }
 
   for (int i = 1; i < argc; i++)
     {
       string arg = argv[i];
@@ -446,28 +448,28 @@ main (int argc, char **argv)
 	  file = arg;
 	  objfiles += (" " + quote_path (arg));
 	}
       else if (ends_with (arg, ".lib") || ends_with (arg, ".a"))
 	{
 	  file = arg;
 	  libfiles += (" " + quote_path (arg));
 	}
-      else if (arg == "-d" || arg == "--debug" || arg == "-v"
-	       || arg == "--verbose")
+      else if (arg == "-d" || arg == "-debug" || arg == "--debug"
+               || arg == "-v" || arg == "-verbose" ||  arg == "--verbose")
 	{
 	  debug = true;
 	  if (vars["CC"] == "cc-msvc")
 	    vars["CC"] += " -d";
 	  if (vars["CXX"] == "cc-msvc")
 	    vars["CXX"] += " -d";
 	  if (vars["DL_LD"] == "cc-msvc")
 	    vars["DL_LD"] += " -d";
 	}
-      else if (arg == "-h" || arg == "-?" || arg == "--help")
+      else if (arg == "-h" || arg == "-?" || arg == "-help" || arg == "--help")
 	{
 	  cout << usage_msg << endl;
 	  cout << help_msg << endl;
 	  return 0;
 	}
       else if (starts_with (arg, "-I"))
 	{
 	  incflags += (" " + quote_path (arg));
@@ -486,64 +488,64 @@ main (int argc, char **argv)
 	{
 	  defs += (" " + arg);
 	}
       else if (starts_with (arg, "-Wl,") || starts_with (arg, "-l")
 	       || starts_with (arg, "-L") || starts_with (arg, "-R"))
 	{
 	  ldflags += (" " + arg);
 	}
-      else if (arg == "-M" || arg == "--depend")
+      else if (arg == "-M" || arg == "-depend" || arg == "--depend")
 	{
 	  depend = true;
 	  compile = false;
 	}
-      else if (arg == "-o" || arg == "--output")
+      else if (arg == "-o" || arg == "-output" || arg == "--output")
 	{
 	  if (i < argc-1)
 	    {
 	      arg = argv[++i];
 	      outputfile = arg;
 	    }
 	  else
 	    cerr << "mkoctfile: output file name missing" << endl;
 	}
-      else if (arg == "-p" || arg == "--print")
+      else if (arg == "-p" || arg == "-print" || arg == "--print")
 	{
 	  if (i < argc-1)
 	    {
 	      arg = argv[++i];
 	      cout << vars[arg] << endl;
 	      return 0;
 	    }
 	  else
 	    cerr << "mkoctfile: --print requires argument" << endl;
 	}
-      else if (arg == "-s" || arg == "--strip")
+      else if (arg == "-s" || arg == "-strip" || arg == "--strip")
 	{
 	  if (no_oct_file_strip_on_this_platform)
 	    cerr << "mkoctfile: stripping disabled on this platform" << endl;
 	  else
 	    strip = true;
 	}
-      else if (arg == "-c" || arg == "--compile")
+      else if (arg == "-c" || arg == "-compile" || arg == "--compile")
 	{
 	  link = false;
 	}
       else if (arg == "-g")
 	{
 	  vars["ALL_CFLAGS"] += " -g";
 	  vars["ALL_CXXFLAGS"] += " -g";
 	  vars["ALL_FFLAGS"] += " -g";
 	}
-      else if (arg == "--link-stand-alone")
+      else if (arg == "-link-stand-alone" || arg == "--link-stand-alone")
 	{
 	  link_stand_alone = true;
 	}
-      else if (arg == "--mex")
+      else if (arg == "-mex" || arg == "--mex")
 	{
 	  incflags += " -I.";
 #ifdef _MSC_VER
 	  ldflags += " -Wl,-export:mexFunction";
 #endif
 	  output_ext = ".mex";
 	}
       else if (starts_with (arg, "-W"))
